contracts/pcv/sentinel/PCVSentinel.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "../../refs/CoreRef.sol";
+import "./IPCVSentinel.sol";
+import "../IPCVDeposit.sol";
+import "../../libs/CoreRefPauseableLib.sol";
+import "./IGuard.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/**
+ * @title PCV Sentinel
+ * @dev the PCV Sentinel should be granted the role Guardian
+ * @notice The PCV Sentinel is a automated extension of the Guardian.
+ * The Guardian can add Guards to the PCV Sentinel. Guards run checks
+ * and provide addresses and calldata for the Sentinel to run, if needed.
+ */
+contract PCVSentinel is IPCVSentinel, CoreRef {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // The set of all guards
+    EnumerableSet.AddressSet private guards;
+
+    constructor(
+        address _core
+    ) CoreRef(_core) {
+        _setContractAdminRole(keccak256("PCV_SENTINEL_ADMIN_ROLE"));
+    }
+
+    // ---------- Read-Only API ----------
+
+    /**
+     * @notice returns whether or not the given address is a guard
+     * @param guard the address to check
+     */
+    function isGuard(address guard) 
+        external 
+        view
+        override
+        returns (bool) 
+    {
+        return guards.contains(guard);
+    }
+
+    /**
+     * @notice returns a list of all guards
+     */
+    function allGuards() 
+        external
+        view
+        override
+        returns (address[] memory all) 
+    {
+        all = new address[](guards.length());
+
+        for (uint i = 0; i < guards.length(); i++) {
+            all[i] = guards.at(i);
+        }
+
+        return all;
```

Comment:
guards.values() should be enough here ?

Reply:
we use openzeppelin v3 not v4 so we don't have access to .values :(

Reply:
whaa really? Thats surprising

Reply:
> whaa really? Thats surprising

yeah I spent a few hours once trying to fix it but that requires using *both* versions in our contract compilation process for anything that depends on core (since we use it in core!) and that became a nightmare

wouldn't recommend it